[Attention OCR] Document model support (#6153)
openvinotoolkit/openvino: docs/MO_DG/prepare_model/convert_model/tf_specific/Convert_AttentionOCR_From_Tensorflow.md

# Convert TensorFlow* Attention OCR Model to Intermediate Representation {#openvino_docs_MO_DG_prepare_model_convert_model_tf_specific_Convert_AttentionOCR_From_Tensorflow}

> This tutorial explains how to convert the Attention OCR (AOCR) model from the [TensorFlow* Attention OCR repository](https://github.com/emedvedev/attention-ocr) to the Intermediate Representation (IR).

Reviewer: please, specify commit hash which you tested your code against

> ## Extract model from `aocr` library

Reviewer 2: ```suggestion
## Extract Model from `aocr` Library
```

> The easiest way to get AOCR model is to download python library `aocr`:

Reviewer 2: ```suggestion
The easiest way to get an AOCR model is to download  `aocr` Python\* library:
```

Author: Done

```
pip install git+https://github.com/emedvedev/attention-ocr.git@master#egg=aocr
```

> This library contains pretrain model and allows to train and run AOCR using command line. After installing `aocr` you can extract model:

Reviewer 2: ```suggestion
This library contains a pretrained model and allows to train and run AOCR using the command line. After installing `aocr`, you can extract the model:
```

```
aocr export --format=frozengraph model/path/
```

> After this step you can find model in model/path/ folder.

Reviewer 2: ```suggestion
After this step, you can find the model in model/path/ folder.
```

## Convert the TensorFlow* AOCR Model to IR

> The original AOCR contains data preprocessing which consists of folowing steps:

Reviewer 2: ```suggestion
The original AOCR contains data preprocessing which consists of the following steps:
```

> The original AOCR contains data preprocessing which consists of the following steps:

Reviewer: `original AOCR contains` -> `original AOCR model contains`

> * Decoding input data to binary format where input data is image as string.

Reviewer 2: where input data is image as string 
maybe 
where input data is an image represented as a string

Author: yes, as you said.

* Resizing binary image to working resolution.

> After that resized image is sent to CNN. The Model Optimizer does not support decoding images so you should exclude preprocessing from model converting. 

Reviewer 2: ```suggestion
After that, the resized image is sent to the convolution neural network (CNN). The Model Optimizer does not support image decoding, so you should exclude preprocessing from model converting. 
```

Author: Done

> After that, the resized image is sent to the convolution neural network (CNN). The Model Optimizer does not support image decoding so you should exclude preprocessing from model converting. 

Reviewer: May be change the final sentence to: `The Model Optimizer does not support image decoding so you should cut-of preprocessing part of the model using "--input" command line parameter`.

```sh
python3 path/to/model_optimizer/mo_tf.py \
--input_model=model/path/frozen_graph.pb \
--input="map/TensorArrayStack/TensorArrayGatherV3:0[1 32 86 1]" \

> --output "transpose_1,transpose_2"

Reviewer: Missing "\\" in the end of the string

--output_dir path/to/ir/
```

Where:
* `map/TensorArrayStack/TensorArrayGatherV3:0[1 32 86 1]` - result of data preprocessing that was cut.

> * `transpose_1,transpose_2` - predictions (int array) and their probabilities (float array).

Reviewer: Change to something like this:
```
* `transpose_1` - name of the node producing tensor with predicted characters.
* `transpose_2` - name of the node producing tensor with predicted characters probabilities.
```
